elide-core/src/main/java/com/yahoo/elide/core/EntityBinding.java

```diff
@@ -188,6 +188,11 @@ private void bindEntityId(Class<?> cls, String type, AccessibleObject fieldOrMet
         String fieldName = getFieldName(fieldOrMethod);
         Class<?> fieldType = getFieldType(fieldOrMethod);
 
+        if (idType != null && fieldType.isAssignableFrom(idType)) {
```

Comment:
If ordering is undefined in the OpenJDK, how can we be sure that we always want the _first_ `idType`? 

This looks like it will cause us to only bind the first field or method annotated with `@Id` we find and not necessarily the `@Id` that is closest to the class we are binding.